packages/client/src/canvasWrapper.ts

```diff
@@ -134,20 +159,32 @@ export class CanvasWrapper {
     }
 
     if (this.isValid) return;
-
+    
     this.clear();
     this.drawDots();
     this.drawRectShapes();
     this._canvasEventHandler.draw(this.context);
     this.isValid = true;
   }
 
-  private drawDots() {
-    const step = 33;
+  scale(factor: number): void {
+    const prevCenter = this.getViewCenter();
+    this.context.scale(factor, factor);
+    const curCenter = this.getViewCenter();
+    
+    const dx = curCenter.x - prevCenter.x
+    const dy = curCenter.y - prevCenter.y
+    this.context.translate(dx, dy);
+  }
+
+  private drawDots(): void {
+    const scale = this.context.getTransform().a;
+    const step = 50;
     for (let x = (step * 3) / 4; x < this.canvas.width; x += step) {
       for (let y = (step * 3) / 4; y < this.canvas.height; y += step) {
+        const pos = this.canvasPointToWorld(x, y);
         this.context.beginPath();
-        this.context.arc(x, y, 1.25, 0, 2 * Math.PI);
+        this.context.arc(pos.x, pos.y, 0.5 / scale, 0, 2 * Math.PI);
```

Comment:
* minor: dots disappear if you zoom too much in;
* minor: I would prefer, if dots were a bit bigger;
* not sure: call to context.arc() throws in Opera